src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,188 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+/**
+ * Parser class which is used to process commands passed down by the user and returns
+ * the result of each command as a String.
+ */
+public class Parser {
+
+    private TaskList list;
+    private boolean isExit;
+
+    public Parser(TaskList list) {
+        this.list = list;
+        this.isExit = false;
+    }
+
+    /**
+     * Checks if the command to exit has been issued
+     * @return True if 'bye' command has been issued, False otherwise.
+     */
+    public boolean isExit() {
+        return isExit;
+    }
+
+    /**
+     * The function that processes the incoming command and executes different
+     * functions depending on the command.
+     * @param command String command provided by the user.
+     * @return String result of the provided command.
+     * @throws DukeException A custom Exception that carries a message for the user if thrown.
+     */
+    public String processCommand(String command) throws DukeException {
```

Comment:
Will it be better to give a more descriptive name for the method, such as identifying the command elements and evaluating the command?